src/Bicep.Cli/Helpers/Deploy/DeploymentProcessor.cs

```diff
@@ -0,0 +1,324 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Text.Json.Nodes;
+using Azure.Core;
+using Azure.Deployments.ClientTools;
+using Azure.Deployments.Core.Definitions;
+using Azure.Deployments.Expression.Intermediate;
+using Azure.Deployments.Expression.Intermediate.Extensions;
+using Azure.Deployments.Templates.Expressions.PartialEvaluation;
+using Azure.ResourceManager;
+using Azure.ResourceManager.Resources;
+using Azure.ResourceManager.Resources.Models;
+using Bicep.Cli.Arguments;
+using Bicep.Core.Emit;
+using Bicep.Core.Extensions;
+using Bicep.Core.Utils;
+using Microsoft.WindowsAzure.ResourceStack.Common.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Bicep.Cli.Commands.Helpers.Deploy;
+
+public record UsingConfig(
+    string? Name,
+    string Scope,
+    StacksConfig? StacksConfig);
+
+public record StacksConfig(
+    string? Description,
+    ActionOnUnmanage? ActionOnUnmanage,
+    DenySettings? DenySettings);
+
+public record DeployCommandsConfig(
+    string Template,
+    string Parameters,
+    UsingConfig UsingConfig);
+
+public class DeploymentProcessor(ArmClient armClient)
+{
+    public static async Task<DeployCommandsConfig> GetDeployCommandsConfig(IEnvironment environment, IReadOnlyDictionary<string, string> additionalArgs, ParametersResult result)
+    {
+        if (result.Template?.Template is not { } template ||
+            result.Parameters is not { } parameters)
+        {
+            throw new Exception($"Failed to compile Bicep parameters");
+        }
+
+        parameters = await ParametersProcessor.Process(parameters, async (kind, config) =>
+        {
+            await Task.CompletedTask;
+            switch (kind)
+            {
+                case "sys.cliArg":
+                    var argKey = config!.Value<string>()!;
+                    var expectedArgument = $"--arg-{argKey}";
+                    return additionalArgs.TryGetValue(argKey) ?? throw new CommandLineException($"CLI argument '{expectedArgument}' must be provided.");
```

Comment:
Are extra `--arg-` CLI arguments (args that do not correspond to a declared external input of type `sys.cliArg`) handled anywhere? Ideally, I think those should raise a CommandLineException

Reply:
Good point - might as well start strict!